MBC_userImport.class.inc

```diff
@@ -419,6 +431,272 @@ class MBC_UserImport
+        $userDetails['birthdate_timestamp'] = $drupalUser['birthdate_timestamp'];
+      }
+      if (isset($drupalUser['phone'])) {
+        $userDetails['mobile'] = $drupalUser['phone'];
+      }
+
+      $payload = serialize($userDetails);
+      $mbUserImport->publishMessage($payload);
+      $status = TRUE;
+
+    }
+    else {
+      $status = FALSE;
+    }
+  }
+
+  /**
+   * Send transactional email to provide details on temporaty password assigned
+   * to newly created Drupal user account.
+   *
+   * @param array $userDetails
+   *   Payload details to be sent with trnsactional message.
+   */
+  private function temporaryPasswordTransactional($userDetails) {
+
+    $config = array(
+      'exchange' => array(
+        'name' => getenv("MB_TRANSACTIONAL_EXCHANGE"),
+        'type' => getenv("MB_TRANSACTIONAL_EXCHANGE_TYPE"),
+        'passive' => getenv("MB_TRANSACTIONAL_EXCHANGE_PASSIVE"),
+        'durable' => getenv("MB_TRANSACTIONAL_EXCHANGE_DURABLE"),
+        'auto_delete' => getenv("MB_TRANSACTIONAL_EXCHANGE_AUTO_DELETE"),
+      ),
+      'queue' => array(
+        'transactional' => array(
+          'name' => getenv("MB_TRANSACTIONAL_QUEUE"),
+          'passive' => getenv("MB_TRANSACTIONAL_QUEUE_PASSIVE"),
+          'durable' => getenv("MB_TRANSACTIONAL_QUEUE_DURABLE"),
+          'exclusive' => getenv("MB_TRANSACTIONAL_QUEUE_EXCLUSIVE"),
+          'auto_delete' => getenv("MB_TRANSACTIONAL_QUEUE_AUTO_DELETE"),
+          'bindingKey' => getenv("MB_TRANSACTIONAL_QUEUE_TOPIC_MB_TRANSACTIONAL_EXCHANGE_PATTERN"),
+        ),
+      ),
+      'routingKey' => 'user.import.transactional',
+    );
+    $mbUserAPI = new MessageBroker($this->credentials, $config);
+
+    $userDetails['activity'] = 'user_password-niche';
+    $userDetails['email_template'] = 'mb-user-signup-niche-com-v1-0-0-1';
+    $userDetails['tags'][0] = 'user_password-' . $userDetails['source'];
+    $payload = serialize($userDetails);
+    $mbUserAPI->publishMessage($payload);
+
+  }
+
+  /**
+   * Sign up user ID (UID) to campaign.
+   *
+   * https://github.com/DoSomething/dosomething/wiki/API#campaign-signup
+   * https://www.dosomething.org/api/v1/campaigns/[nid]/signup
+   *
+   * @param integer $campaignNID
+   *   The unique node ID from the Drupal site that identifies the target
+   *   campaign to sign the user up to.
+   * @param array $drupalUID
+   *   The user ID (UID) of the Drupal user account to create a campaign
+   *   signup for.
+   * @param string $source
+   *   The name of the import source
+   *
+   * @return string $status
+   *   Results for transactional submission.
+   */
+  private function campaignSignup($campaignNID, $drupalUID, $source) {
+
+    $post = array(
+      'source' => $source . '_mb_import',
+      'uid' => $drupalUID
+    );
+    $curlUrl = $this->settings['ds_drupal_api_host'];
+    $port = $this->settings['ds_drupal_api_port'];
+    if ($port != 0) {
+      $curlUrl .= ":{$port}";
```

Comment:
Good point! With the double quotes '"", `$port` will get replaced with the value to create the string ":80".